modules/meteroid/crates/meteroid-store/src/domain/invoices.rs

```diff
@@ -262,6 +289,63 @@ impl InvoiceTotals {
             total,
             tax_amount,
             applied_credits,
+            applied_coupons: coupons_discount.applied_coupons,
+        }
+    }
+
+    fn calculate_coupons_discount(
+        subtotal: i64,
+        invoice_currency: &str,
+        coupons: &[AppliedCouponDetailed],
+    ) -> AppliedCouponsDiscount {
+        let applicable_coupons: Vec<&AppliedCouponDetailed> = coupons
+            .iter()
+            .filter(|x| x.is_invoice_applicable())
+            .sorted_by_key(|x| x.applied_coupon.created_at)
+            .collect::<Vec<_>>();
+
+        let mut applied_coupons: BTreeMap<Uuid, i64> = BTreeMap::new();
+
+        let mut subtotal_subunits = Decimal::from(subtotal);
+
+        for applicable_coupon in applicable_coupons {
+            if subtotal_subunits <= Decimal::ONE {
+                break;
+            }
+            let discount = match &applicable_coupon.coupon.discount {
+                CouponDiscount::Percentage(percentage) => {
+                    subtotal_subunits * percentage / Decimal::ONE_HUNDRED
+                }
+                CouponDiscount::Fixed { amount, currency } => {
+                    // todo currency conversion
+                    if currency != invoice_currency {
+                        continue;
+                    }
```

Comment:
can we check it & reject when applying the coupon to the subscription for transparency ?

Reply:
fixed